### DISCUSSION TITLE
Printing

### DISCUSSION THREAD

**Discussion by StuEy40** [QuestPDF/QuestPDF#421] [Q&A] [answered]
Hi,

Thank you for a wonderful package. I am using it successfully and it has been a great help for an in-house application. Can you please pass on any advice regarding printing a generated document.

I have succeeded in sending the byte array generated by .GeneratePdf() directly to a printer in these two ways. 

1. Using a TcpClient, client.Connect(printerIP, 9100), and a NetworkStream, stream.Write(docket, 0, docket.Length), to send the array to the printer's IP address. Note that 'docket' in this code is the byte array.
2. By using File.Copy("myfile.pdf", @"\\server\printerShareName"). The file was created and stored on disk first using .GeneratePdf("myfile.pdf")

Both of these approaches work well but I cannot choose the output tray of the destination printer using these methods.

Has anyone successfully used the System.Drawing.Printing namespace in WinForms to print a QuestPDF document, which allows you to choose the output tray on the printer? All the examples typically use a StreamReader object and only show how to print a plain text file from the stream, not a byte array.

Thanks for any advice.

Regards,
Stu.

**Comment by girlpunk** [comment, marked as answer]
Hi Stu, This is rather out of scope for QuestPDF, so I'm not too sure on the `System.Drawing.Printing` classes (though it's likely you're along the right lines). As a suggestion, you may want to look at CUPS as an alternative if this doesn't prove feasible directly.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. As @girlpunk mentioned, printing and printer tray selection are outside the scope of QuestPDF, which focuses on PDF document generation. We hope you were able to find a solution.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
